clients/filesystem-hadoop/build.gradle.kts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+plugins {
+  `maven-publish`
+  id("java")
+}
+
+dependencies {
+  compileOnly(libs.hadoop2.common)
```

Comment:
We use `compileOnly` here for `hadoop-common`, and import `client-java-runtime` in shadow, maybe we do not need to create a shaded `filesystem-hadoop` module extraly.

Reply:
How do you use it if there's no runtime jar? For example, like Spark, you can only use it like `--packages`, not `--jars`, am I right?

Reply:
Yes, the current packaging method is similar to `trino-connector`, we will put all dependencies into the `/build/libs` directory of the `filesystem-hadoop` module. Like this:  
![2a6b9297-9a61-4e19-b439-e53b7738cf16](https://github.com/datastrato/gravitino/assets/26177232/2fc9ff49-1740-4665-a73a-184cac68b0e9)
When users use Spark to read and write the path of `gvfs`, they need to put these dependencies into the `/jars` directory of Spark Pack. My consideration is whether we still need to create a `filesystem-hadoop-runtime` module for `filesystem-hadoop` to shade, because in the current dependencies, `hadoop-common` will not directly participate in packaging, and `client-java-runtime` itself is already used shaded mode introduced

Reply:
Let me think a bit on this. It is unrelated to this PR, we can do this in another PR if required.

Reply:
Add the `filesystem-hadoop-runtime` module for users, so they can only import the `filesystem-hadoop-runtime.jar`.
![image](https://github.com/datastrato/gravitino/assets/26177232/8c24d2a5-11a4-4987-8cca-1cc13e49ae37)